Library/Homebrew/download_strategy.rb

```diff
@@ -423,6 +423,8 @@ def _curl_args
 
     args += ["--header", meta.fetch(:header)] if meta.key?(:header)
 
+    meta.fetch(:headers).each { |h| args += ["--header", h.strip] } if meta.key?(:headers)
+
```

Comment:
How about something like:

```suggestion
    args += [meta.fetch(:header), *meta.fetch(:headers)].compact.flat_map {|h| ["--header", h.strip]}

```

Note I haven't tested this locally and it might need tweaks to pass `brew style`.

Reply:
That looks better, I've added it and ran it through style

Reply:
@MikeMcQuaid Ahh its a bit tricky because you need to check for the keys existing before you use meta.fetch

I have a solution like
```
    header_keys = []
    header_keys << meta.fetch(:header) if meta.key?(:header)
    header_keys << meta.fetch(:headers) if meta.key?(:headers)
    args += header_keys.compact.flatten.flat_map { |h| ["--header", h.strip] }
```

But I don't know if that's better than the original (Also the double flatten and flat map, but it appears to need two passes of flattening to work which isn't exposed by flat_map)

Reply:
Alternatively, `meta.fetch(:header, nil)` or even just simpler `meta[:header]` allow the original approach to be used

Reply:
@MikeMcQuaid Sorry for the delay, I've taken your advice and updated the PR